### PR TITLE
Fix lexical-dynamic warnings

### PR DESCRIPTION
**Pull request by matsl**
## What

Fix lexical-dynamic warnings.

## Why

```
kotl/kview.el:807:54: Warning: Lexical argument shadows the dynamic variable fill-prefix
kotl/kview.el:1399:10: Warning: Variable ‘pos’ left uninitialized
hmouse-sh.el:276:29: Warning: Lexical argument shadows the dynamic variable hmouse-middle-flag
hmouse-sh.el:472:30: Warning: Lexical argument shadows the dynamic variable hmouse-middle-flag
hpath.el:1656:27: Warning: Variable ‘modifier’ left uninitialized
```

### CODE REVIEW

File: hmouse-sh.el [hunk 1]
```diff
@@ -273,15 +273,15 @@ This includes depresses and drags.  Shifted Mouse Key 1 is
 	   [mode-line S-drag-mouse-5]
 	   [mode-line S-mouse-5])))))
 	   
-(defun hmouse-get-bindings (hmouse-middle-flag)
+(defun hmouse-get-bindings (middle-flag)
```
Comment (matsl) [inline hmouse-sh.el:276]: Parameter has same name as global dynamic variable. All instances calls the method with the arg `fill-prefix`

File: kotl/kview.el [hunk 2]
```diff
@@ -804,10 +804,10 @@ On success, return t, else nil."
     (when (kotl-mode:goto-cell permanent-id)
       (kcell-view:label))))
 
-(defun kview:insert-contents (kcell contents no-fill fill-prefix)
+(defun kview:insert-contents (kcell contents no-fill cell-fill-prefix)
```
Comment (matsl) [inline kotl/kview.el:807]: Global dynamic variable name used as parameter.

File: hpath.el [hunk 3]
```diff
@@ -1653,7 +1653,7 @@ form is what is returned for PATH."
 		(file-readable-p path))
       (setq path (hpath:call
 		  (lambda (path non-exist)
-		    (let (modifier
+		    (let ((modifier nil)
```
Comment (matsl) [inline hpath.el:1656]: @rswgnu `modifier` is strange. Never set and used in call to two later `concat` calls. Code might be possible to clean up more so leaving this for you to look into if `modifier` can be removed and the `concat` calls simplified?

File: kotl/kview.el [hunk 4]
```diff
@@ -1395,8 +1395,7 @@ new outlines is also set to this new value."
 				  (- sep-len-increase) ?\ )))
 		   (while (re-search-forward indent nil t)
 		     (delete-region
-		      (+ (match-beginning 0) 2) (match-end 0)))))))
-	 pos)
```
Comment (matsl) [inline kotl/kview.el:1399]: pos does not look like it is used so removing it.

Comment (rswgnu) [reply kotl/kview.el:1399]: You are right that it is not really used but it is sent in a call about 2 lines down but then the value is ignored in the called function as well.  Let's push your updates and then I'll finish the changes here after.

**Review by rswgnu — APPROVED** [review]

**Pull request merged**